tfhe-ntt/src/prime32.rs

```diff
@@ -590,10 +629,22 @@ fn mul_accumulate_scalar(
         let c1 = (d >> big_q_m1) as u32;
         let c3 = ((c1 as u64 * p_barrett as u64) >> 32) as u32;
         let prod = (d as u32).wrapping_sub(p.wrapping_mul(c3));
-        let prod = prod.min(prod.wrapping_sub(p));
+        let mut prod = prod;
+        if prod >= p {
+            prod = prod.wrapping_sub(p);
+            if prod >= p {
+                prod = prod.wrapping_sub(p);
+            }
+        }
 
-        let acc_ = prod + *acc;
-        *acc = acc_.min(acc_.wrapping_sub(p));
+        let mut acc_ = prod + *acc;
+        if acc_ >= p {
+            acc_ = acc_.wrapping_sub(p);
+            if acc_ >= p {
+                acc_ = acc_.wrapping_sub(p);
+            }
+        }
+        *acc = acc_;
```

Comment:
I'm sorry to say this code does not fix the case reported in the original issue.

If I don't end up re-using code from this PR (which does not seem to provide a fix after all) I will not be crediting this work